Assets/HoloToolkit/Input/Scripts/Voice/KeywordManager.cs

```diff
@@ -44,23 +44,36 @@ public enum RecognizerStartBehavior { AutoStart, ManualStart };
         public KeywordAndResponse[] KeywordsAndResponses;
 
         private KeywordRecognizer keywordRecognizer;
-        private Dictionary<string, UnityEvent> responses;
+        private readonly Dictionary<string, UnityEvent> responses = new Dictionary<string, UnityEvent>();
 
         void Start()
         {
-            if (KeywordsAndResponses.Length > 0)
+            int keywordCount = KeywordsAndResponses.Length;
+            if (keywordCount > 0)
             {
-                // Convert the struct array into a dictionary, with the keywords and the keys and the methods as the values.
-                // This helps easily link the keyword recognized to the UnityEvent to be invoked.
-                responses = KeywordsAndResponses.ToDictionary(keywordAndResponse => keywordAndResponse.Keyword,
-                                                              keywordAndResponse => keywordAndResponse.Response);
-
-                keywordRecognizer = new KeywordRecognizer(responses.Keys.ToArray());
-                keywordRecognizer.OnPhraseRecognized += KeywordRecognizer_OnPhraseRecognized;
-
-                if (RecognizerStart == RecognizerStartBehavior.AutoStart)
+                try
                 {
-                    keywordRecognizer.Start();
+                    string[] keywords = new string[keywordCount];
+                    // Convert the struct array into a dictionary, with the keywords and the keys and the methods as the values.
+                    // This helps easily link the keyword recognized to the UnityEvent to be invoked.
+                    for (int index = keywordCount; --index >= 0;)
```

Comment:
Is LINQ "bad" enough to remove it from a method that only gets called once (or maybe a handful of times, if you have more than one KeywordManager) in an app's lifetime? This code seems slightly less readable to me, though I'm all good with replacing this if I have a minority opinion!